template/App.tsx

```diff
@@ -56,43 +56,56 @@ function Section({children, title}: SectionProps): React.JSX.Element {
 }
 
 function App(): React.JSX.Element {
+  return (
+    <SafeAreaProvider>
+      <MainContent />
+    </SafeAreaProvider>
+  );
+}
+
+function MainContent(): React.JSX.Element {
+  const insets = useSafeAreaInsets();
+
   const isDarkMode = useColorScheme() === 'dark';
 
   const backgroundStyle = {
     backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
+    bottomPadding: insets.bottom,
   };
 
+  const contentContainerStyle = { paddingBottom: insets.bottom };
+
   return (
-    <SafeAreaView style={backgroundStyle}>
-      <StatusBar
-        barStyle={isDarkMode ? 'light-content' : 'dark-content'}
-        backgroundColor={backgroundStyle.backgroundColor}
-      />
-      <ScrollView
-        contentInsetAdjustmentBehavior="automatic"
-        style={backgroundStyle}>
-        <Header />
-        <View
-          style={{
-            backgroundColor: isDarkMode ? Colors.black : Colors.white,
-          }}>
-          <Section title="Step One">
-            Edit <Text style={styles.highlight}>App.tsx</Text> to change this
-            screen and then come back to see your edits.
-          </Section>
-          <Section title="See Your Changes">
-            <ReloadInstructions />
-          </Section>
-          <Section title="Debug">
-            <DebugInstructions />
-          </Section>
-          <Section title="Learn More">
-            Read the docs to discover what to do next:
-          </Section>
-          <LearnMoreLinks />
-        </View>
-      </ScrollView>
-    </SafeAreaView>
+    <View style={backgroundStyle}>
+        <StatusBar
+          barStyle={isDarkMode ? 'light-content' : 'dark-content'}
+          backgroundColor={backgroundStyle.backgroundColor}
+        />
+        <ScrollView
+          contentContainerStyle={contentContainerStyle}
+          style={backgroundStyle}>
+          <Header badgeTopMargin={insets.top}/>
```

Comment:
passing in `badgeTopMargin` to the `HermesBadge` within `Header` has proper margin to avoid overlap with top status bar.

Pre-requisite: This code requires updated to react-native with this PR applied: https://github.com/facebook/react-native/pull/47708